.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,34 @@
+name: Pull Request
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 7.0.x
+    - name: Check code formatting
+      run: |
+        dotnet tool restore
+        dotnet fantomas --check -r src samples
+
+  avalonia:
+    name: Fabulous.Avalonia
+    runs-on: macos-latest
```

Comment:
I would be consistent here and use the same runner between all the files.
`macos-12` should be good